pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -234,20 +238,17 @@ func (nc *mockNotificationChannel) totalNotifications() int {
 	return total
 }
 
-func (nc *mockNotificationChannel) matchesExpNotifications(exp map[string][]string) bool {
+func (nc *mockNotificationChannel) matchesExpNotifications(t *testing.T, exp map[string][]string) {
```

Comment:
Should you rewrite the method to use `t` instead of `nc.t`? I think it will be safer in any case, in case of being called from a subtest with a different `t` than `nc.t`.